flexmeasures/data/models/reporting/tests/test_tibber_reporter.py

```diff
@@ -0,0 +1,252 @@
+    115.04,
+    122.15,
+    105.39,
+    83.40,
+    34.90,
+    -4.50,
+    -50.00,
+    -50.07,
+    -50.00,
+    -0.90,
+    08.10,
+    128.10,
+    151.00,
+    155.20,
+    152.00,
+    134.04,
+    120.10,
+]  # EUR / MWh
+
+tibber_app_price = [
+    29.2,
+    27.7,
+    27.0,
+    26.2,
+    26.0,
+    27.4,
+    29.8,
+    31.3,
+    32.2,
+    30.2,
+    27.5,
+    21.7,
+    16.9,
+    11.4,
+    11.4,
+    11.4,
+    17.3,
+    30.5,
+    32.9,
+    35.7,
+    36.2,
+    35.8,
+    33.6,
+    32.0,
+]  # EUR/MWh
+
+
+class TibberReporter(PandasReporter):
+    def __init__(self, start: datetime, end: datetime) -> None:
+        """This class calculates the price of energy of a tariff indexed to the Day Ahead prices.
+        Energy Price = (1 + VAT) x ( EB + Tiber + DA Prices)
+        """
+
+        # search the sensors
+        EB = Sensor.query.filter(Sensor.name == "EB").one_or_none()
+        BWV = Sensor.query.filter(Sensor.name == "BWV").one_or_none()
+        tibber_tariff = Sensor.query.filter(
+            Sensor.name == "Tibber Tariff"
+        ).one_or_none()
+
+        da_prices = Sensor.query.filter(Sensor.name == "DA prices").one_or_none()
+
+        tb_query_config_extra = dict(
+            resolution=3600,  # 1h = 3600s
+            event_starts_after=str(start),
+            event_ends_before=str(end),
+        )
+
+        # creating the PandasReporter reporter config
+        reporter_config_raw = dict(
+            start=str(start),
+            end=str(end),
+            tb_query_config=[
+                dict(sensor=EB.id, **tb_query_config_extra),
+                dict(sensor=BWV.id, **tb_query_config_extra),
+                dict(sensor=tibber_tariff.id, **tb_query_config_extra),
+                dict(sensor=da_prices.id, **tb_query_config_extra),
+            ],
+            transformations=[
+                dict(
+                    df_input="sensor_1",
+                    df_output="BWV",
+                    method="droplevel",
+                    args=[[1, 2, 3]],
+                ),
+                dict(method="add", args=[1]),  # this is to get 1 + BWV
+                dict(
+                    df_input="sensor_2",
+                    df_output="EB",
+                    method="droplevel",
+                    args=[[1, 2, 3]],
+                ),
+                dict(
+                    df_input="sensor_3",
+                    df_output="tibber_tariff",
+                    method="droplevel",
+                    args=[[1, 2, 3]],
+                ),
+                dict(
+                    df_input="sensor_4",
+                    df_output="da_prices",
+                    method="droplevel",
+                    args=[[1, 2, 3]],
+                ),
+                dict(
+                    method="multiply",
+                    args=[1 / 1000],
+                ),
+                dict(
+                    df_output="energy_price",
+                    df_input="EB",
```

Comment:
I would find it more intuitive to start with the day-ahead price and to add taxes to it, rather than the other way around. I know it doesn't matter to the result, but it makes the reporter_config easier to read.

Reply:
I agree. Now it starts with the day-ahead prices.